gremlin-go/driver/protocol.go

```diff
@@ -123,10 +123,33 @@ func responseHandler(resultSets map[string]ResultSet, response response, log *lo
 		// Add data to the ResultSet.
 		resultSets[responseIDString].addResult(&Result{data})
 		log.logger.Logf(Info, "Partial %v===>%v", response.responseStatus, data)
-	} else if statusCode == http.StatusProxyAuthRequired {
-		// TODO AN-989: Implement authentication (including handshaking).
-		resultSets[responseIDString].Close()
-		return errors.New("authentication is not currently supported")
+	} else if statusCode == http.StatusProxyAuthRequired || statusCode == 151 {
+		// http status code 151 is not defined here, but corresponds with 403, i.e authetication has failed.
+		// Server has requested basic auth.
+		authInfo := protocol.transporter.getAuthInfo()
+		if authInfo.getUseBasicAuth() {
+			username := []byte(authInfo.Username)
+			password := []byte(authInfo.Password)
+
+			authBytes := make([]byte, 0)
+			authBytes = append(authBytes, 0)
+			for _, c := range username {
+				authBytes = append(authBytes, c)
+			}
```

Comment:
not sure it's better for golang, but shorter
```suggestion
			authBytes = append(authBytes, []byte(username)...)
```

Reply:
I didn't know you could do that - thanks!